### DISCUSSION TITLE
Debug Code

### DISCUSSION THREAD

**Discussion by x4code** [ExpressLRS/ExpressLRS#2202] [General]
Hello,
is it posible to know how you debug code? 
how do i read this DBGLN messages?
Thank you.

Regads,
Filipe

**Comment by pkendall64** [comment]
Compile the firmware in platformio and set the option `-DDEBUG_LOG` in super_defines.txt (a new file in the same folder as `user_defines.txt`) You will also see there are other options in `user_defines.txt` that you can copy over to `super_defines.txt` to control the output. The debug messages will be mixed into the serial output on a receiver and on a TX module you can set some pins for the debug serial port.

**Reply by x4code** [reply]
thanks a lot mate

**Reply by tech-support-ah-oss** [reply]
pio device monitor -p /dev/cu.usbserial-0001 -b 115200

--- forcing DTR inactive
--- forcing RTS inactive
--- Terminal on /dev/cu.usbserial-0001 | 115200 8-N-1
--- Available filters and text transformations: debug, default, direct, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
ets Jun  8 2016 00:22:57

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13232
load:0x40080400,le

can you help me ?
